### PR TITLE
Feature/io 421 base architecture

### PR DESCRIPTION
**Pull request by nikolay-ahramovich-idf**

### CODE REVIEW

File: packages/data/.metadata [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# This file tracks properties of this Flutter project.
+# Used by Flutter tool to assess capabilities and perform upgrades etc.
+#
+# This file should be version controlled and should not be manually edited.
+
+version:
+  revision: f1875d570e39de09040c8f79aa13cc56baab8db1
```
Comment (ekaterina-demeshko-idf) [inline packages/data/.metadata]: gitignore it

File: packages/domain/.metadata [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# This file tracks properties of this Flutter project.
+# Used by Flutter tool to assess capabilities and perform upgrades etc.
+#
+# This file should be version controlled and should not be manually edited.
+
+version:
```
Comment (ekaterina-demeshko-idf) [inline packages/domain/.metadata]: -//-

File: packages/presentation/.metadata [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# This file tracks properties of this Flutter project.
+# Used by Flutter tool to assess capabilities and perform upgrades etc.
+#
+# This file should be version controlled and should not be manually edited.
+
+version:
+  revision: ffccd96b62ee8cec7740dab303538c5fc26ac543
+  channel: stable
```
Comment (ekaterina-demeshko-idf) [inline packages/presentation/.metadata]: -//-

**Review by ekaterina-demeshko-idf — APPROVED** [review]

File: packages/data/.metadata (hunk 1 above)
Comment (nikolay-ahramovich-idf) [reply packages/data/.metadata]: fixed.

File: packages/domain/.metadata (hunk 2 above)
Comment (nikolay-ahramovich-idf) [reply packages/domain/.metadata]: fixed.

File: packages/presentation/.metadata (hunk 3 above)
Comment (nikolay-ahramovich-idf) [reply packages/presentation/.metadata]: fixed.

File: .gitignore [hunk 4]
```diff
@@ -31,7 +31,8 @@ migrate_working_dir/
 .pub-cache/
 .pub/
 /build/
-/pubspec.lock
+**/pubspec.lock
```
Comment (sergei-mikhailovskii-idf) [inline .gitignore:34]: top solution!

File: packages/presentation/lib/bloc/base/bloc.dart [hunk 5]
```diff
@@ -0,0 +1,8 @@
+abstract class Bloc<State> {
+  initState();
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/bloc/base/bloc.dart]: add return type

File: packages/presentation/lib/bloc/base/bloc.dart [hunk 6]
```diff
@@ -0,0 +1,8 @@
...
+  State? get state;
+  set state(State? newState);
+  Stream<State> get stream;
+  void add(dynamic data);
+  void dispose();
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/bloc/base/bloc.dart]: formatting

variable
variable

method
method

File: packages/presentation/lib/star_movie_app.dart [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      home: Navigator(
+        pages: [
+          SplashScreen.page(),
+        ],
+        onPopPage: (route, result) => route.didPop(result),
+      ),
+    );
+  }
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/star_movie_app.dart]: need to add navigator logic

**Review by sergei-mikhailovskii-idf — CHANGES_REQUESTED** [review]

File: packages/presentation/lib/star_movie_app.dart (hunk 7 above)
Comment (nikolay-ahramovich-idf) [reply packages/presentation/lib/star_movie_app.dart]: Fixed.

File: packages/presentation/lib/bloc/base/bloc.dart (hunk 6 above)
Comment (nikolay-ahramovich-idf) [reply packages/presentation/lib/bloc/base/bloc.dart]: Fixed.

File: packages/presentation/lib/bloc/base/bloc.dart (hunk 5 above)
Comment (nikolay-ahramovich-idf) [reply packages/presentation/lib/bloc/base/bloc.dart]: Fixed.

File: packages/presentation/lib/bloc/base/bloc.dart [hunk 8]
```diff
@@ -0,0 +1,9 @@
+abstract class Bloc<S> {
+  S get state;
+  set state(S newState);
+  Stream<S?> get stream;
+
+  void initState();
+  void add(dynamic data);
```
Comment (sergei-mikhailovskii-idf) [inline packages/presentation/lib/bloc/base/bloc.dart]: dynamic -> S

**Review by sergei-mikhailovskii-idf — APPROVED** [review]

File: packages/presentation/lib/bloc/base/bloc.dart (hunk 8 above)
Comment (nikolay-ahramovich-idf) [reply packages/presentation/lib/bloc/base/bloc.dart]: Fixed.

**Review by andrei-lastochkin-idf — APPROVED** [review]

**Pull request merged**